Detectors/EMCAL/workflow/src/StandaloneAODProducerSpec.cxx

```diff
@@ -0,0 +1,171 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "EMCALWorkflow/StandaloneAODProducerSpec.h"
+#include "Framework/TableBuilder.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/DataTypes.h"
+#include "Framework/InputRecordWalker.h"
+#include "Framework/Logger.h"
+#include "Framework/TableBuilder.h"
+#include "Framework/TableTreeHelpers.h"
+#include "MathUtils/Utils.h"
+
+using namespace o2::framework;
+
+namespace o2
+{
+namespace emcal
+{
+
+StandaloneAODProducerSpec::StandaloneAODProducerSpec()
+{
+  mTimer.Stop();
+  mTimer.Reset();
+}
+
+void StandaloneAODProducerSpec::init(o2::framework::InitContext& ic)
+{
+  mCaloEventHandler = new o2::emcal::EventHandler<o2::emcal::Cell>();
+  mCaloAmp = 0xFFFFFFFF;
+  mCaloTime = 0xFFFFFFFF;
+
+  mTFNumber = ic.options().get<int64_t>("aod-timeframe-id");
+  mRunNumber = ic.options().get<int>("run-number");
+}
+
+void StandaloneAODProducerSpec::run(ProcessingContext& pc)
+{
+  auto cput = mTimer.CpuTime();
+  mTimer.Start(false);
+
+  const auto* dh = o2::header::get<o2::header::DataHeader*>(pc.inputs().getFirstValid(true).header);
+  uint64_t tfNumber;
+  if (mTFNumber == -1L) {
+    // TODO has to be made globally unique (by using absolute time of TF). For now is unique within the run
+    tfNumber = dh->tfCounter; // getTFNumber(startIR, runNumber);
```

Comment:
Note that this part was changed in the standard AOD producer: https://github.com/AliceO2Group/AliceO2/blob/c394a041e0f17a0fd6971b8373597d2e407a2e13/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L1101

I'll fix this and merge

Reply:
Thanks a lot for fixing!